probe-rs/src/session.rs

```diff
@@ -243,11 +243,12 @@ impl Session {
             }
         }
 
-        let interface = probe.try_into_arm_interface().map_err(|(_, err)| err)?;
+        let mut interface = probe
+            .try_into_arm_interface(sequence_handle.clone())
+            .map_err(|(_, err)| err)?;
+
+        interface.select_debug_port(default_dp)?;
```

Comment:
I am wondering if we could make this mandatory through the API somehow. For internal use like here it's fine, but for the enduser it's kind of bad to require this implicitly.